### DISCUSSION TITLE
Interesting, needs explanation though

### DISCUSSION THREAD

**Discussion by oathzed** [MunifTanjim/stremthru#3] [Q&A]
This looks interesting, but I'm not quite sure what exactly it does. It would be great if you added a little bit explaining about what this actually does and it's use cases. Thank you!

**Comment by MunifTanjim** [comment]
Think of StremThru as an interface to Debrid services.

If you write a Stremio Addon and need to integrate Debrid services, you can write integration with StremThru and gain support for all the Debrid service StremThru supports (currently only AllDebrid... RealDebrid coming next).

If you're interested, I just opened a PR for Comet: https://github.com/g0ldyy/comet/pull/166

StremThru also supports proxy streaming. When you're streaming files (videos) on Stremio, all the traffic will go through StremThru, so you won't be directly accessing the links you get from Debrid services.

Hope that clear things up.

**Reply by IamJanarthan** [reply]
How to do the proxy streaming part? is there any tutorial for it?

**Reply by MunifTanjim** [reply]
You need to self-host it and set the `STREMTHRU_PROXY_AUTH` and `STREMTHRU_STORE_AUTH` config.

You can join the discord community to get help from other self-hosters: https://discord.com/channels/1148672013028831352/1323628744723660851

**Reply by IamJanarthan** [reply]
Any other link? It's not working for me

**Reply by MunifTanjim** [reply]
Try this: https://discord.gg/rivenmedia

**Comment by MunifTanjim** [comment]
I'm not familiar with rlsbb. Can you share more details, please? Is it another debrid service provider?

**Comment by oathzed** [comment]
To add to that, you put the rapidgator link to all debrids downloader and to real debrids download page, and it creates a direct link for you. Both have apis for it

**Comment by MunifTanjim** [comment]
I see. But I'm confused about how that'll be useful for Stremio. Is there any Stremio addon that'll benefit from it? How would the integration look like?

N.B. StremThru itself is not a Stremio addon.
